docs/guides/lint.md

```diff
@@ -3,75 +3,235 @@ redirectFrom:
+        - # header configuration
+```
+
+#### Header configuration
+
+| Property      | Description | Examples |
+| ------------- | ----------- | -------- |
+| `matches`     | Glob match against the URL. | `https://api.example.com/**` or `https://api.example.com/*.json` |
+| `name`        | HTTP header name.           | `Authorization` or `X-API-KEY` |
+| `value`       | The value of the header. Mutually exclusive with `envVariable`. | `123-abc` |
+| `envVariable` | The name of the environment variable that contains the value of the header. Mutually exclusive with `value`. | `SECRET_KEY` |
+
+:::success
+It is recommeded to use environment variables where possible.
+:::
+
+#### Example
+
+```yaml multiple values
+lint:
+  resolve:
+    http:
+      headers:
+        - matches: https://api.example.com/v2/**
+          name: X-API-KEY
+          envVariable: SECRET_KEY
+        - matches: https://example.com/*/test.yaml
+          name: Authorization
+          envVariable: SECRET_AUTH
+```
+
+The first match will win in the event when a URL matches multiple patterns. Therefore, only the header from the first match will be used in the request.
+
+### Preprocessors
+
+Use this section to change the [severity level](#severity-levels) of any rules in your extended configurations. Some rules may also receive [additional configurations](#additional-rule-options).
+
+* **type**: `array of objects`
+* **possible values**: `error`, `warn`, `off`
+
+Preprocessors run first during `lint` and `bundle`.
+
+:::info
+Linting happens only when the `--lint` flag is supplied
+:::
+
+:::info
+Preprocessors are rarely indicated - avoid if possible.
+
+This section can be omitted.
+:::
+
+### Rules
+
+Use this section to change the [severity level](#severity-levels) of any rules in your extended configurations. Some rules may also receive [additional configurations](#additional-rule-options).
+
+* **type**: `array of objects`
+* **possible values**: `error`, `warn`, `off`
+
+For `lint` command: rules run *after* preprocessors.
+For `bundle` command: rules run *between* preprocessors and decorators.
+
+#### Examples
 
-  preprocessors:
-    # Preprocessors are rarely indicated -- avoid if possible.
-    # This section can be omitted.
+```yaml short version
+lint:
+  rules:
+    no-sibling-refs: error
+```
 
+```yaml verbose version
+lint:
   rules:
-    # Override any rules that are available from the extends configuration.
-    # Severity options are "error", "warn" or "off".
-    no-sibling-refs: error # Uses syntactic sugar to define the severity level of an option.
-    # Alternative verbose configuration
     no-sibling-refs:
       severity: error
-    # Some rules may have additional configuration options. Use the verbose configuration style in those cases.
-    # This boolean parameter prefixes example overrides the default "prefixes".
+```
+
+```yaml rules with additional configuration
+# Use verbose configuration version to define additional configuration
```

Comment:
```suggestion
# Use verbose configuration syntax to define additional configuration
```

Reply:
"style" could also work here if "syntax" is too technical. "Version" can be misleading so I would avoid it